storage/pebble/bootstrap_test.go

```diff
@@ -0,0 +1,249 @@
+		checkpointFile := path.Join(dir, fileName)
+		pb, dbDir := createPebbleForTest(t)
+		bootstrap, err := NewRegisterBootstrap(pb, checkpointFile, rootHeight, log)
+		require.NoError(t, err)
+		err = bootstrap.IndexCheckpointFile(context.Background())
+		require.NoError(t, err)
+
+		// create registers instance and check values
+		reg, err := NewRegisters(pb)
+		require.NoError(t, err)
+
+		require.Equal(t, reg.LatestHeight(), rootHeight)
+		require.Equal(t, reg.FirstHeight(), rootHeight)
+
+		for _, register := range registerIDs {
+			val, err := reg.Get(*register, rootHeight)
+			require.NoError(t, err)
+			require.Equal(t, val, []byte{defaultRegisterValue})
+		}
+
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
+	})
+
+}
+
+func simpleTrieWithValidRegisterIDs(t *testing.T) ([]*trie.MTrie, []*flow.RegisterID) {
+	return trieWithValidRegisterIDs(t, 2)
+}
+
+func largeTrieWithValidRegisterIDs(t *testing.T) ([]*trie.MTrie, []*flow.RegisterID) {
+	// large enough trie so every worker should have something to index
+	largeTrieSize := 2 * pebbleBootstrapRegisterBatchLen * pebbleBootstrapWorkerCount
+	return trieWithValidRegisterIDs(t, uint16(largeTrieSize))
+}
+
+func trieWithValidRegisterIDs(t *testing.T, n uint16) ([]*trie.MTrie, []*flow.RegisterID) {
+	emptyTrie := trie.NewEmptyMTrie()
+	resultRegisterIDs := make([]*flow.RegisterID, 0, n)
+	paths := randomRegisterPaths(n)
+	payloads := randomRegisterPayloads(n)
+	for _, payload := range payloads {
+		key, err := payload.Key()
+		require.NoError(t, err)
+		regID, err := convert.LedgerKeyToRegisterID(key)
+		require.NoError(t, err)
+		resultRegisterIDs = append(resultRegisterIDs, &regID)
+	}
+	populatedTrie, depth, err := trie.NewTrieWithUpdatedRegisters(emptyTrie, paths, payloads, true)
+	// make sure it has at least 1 leaf node
+	require.GreaterOrEqual(t, depth, uint16(1))
+	require.NoError(t, err)
+	resultTries := []*trie.MTrie{emptyTrie, populatedTrie}
+	return resultTries, resultRegisterIDs
+}
+
+func randomRegisterPayloads(n uint16) []ledger.Payload {
+	p := make([]ledger.Payload, 0, n)
+	for i := uint16(0); i < n; i++ {
+		o := make([]byte, 0, 8)
+		o = binary.BigEndian.AppendUint16(o, n)
+		k := ledger.Key{KeyParts: []ledger.KeyPart{
+			{Type: convert.KeyPartOwner, Value: o},
+			{Type: convert.KeyPartKey, Value: o},
+		}}
+		// values are always 'v' for ease of testing/checking
+		v := ledger.Value{defaultRegisterValue}
+		pl := ledger.NewPayload(k, v)
+		p = append(p, *pl)
+	}
+	return p
+}
+
+func randomRegisterPaths(n uint16) []ledger.Path {
+	p := make([]ledger.Path, 0, n)
+	for i := uint16(0); i < n; i++ {
+		p = append(p, testutils.PathByUint16(i))
+	}
+	return p
+}
+
+func createPebbleForTest(t *testing.T) (*pebble.DB, string) {
+	dbDir := unittest.TempPebblePath(t)
+	pb, err := OpenRegisterPebbleDB(dbDir)
+	require.NoError(t, err)
+	return pb, dbDir
```

Comment:
optional: consider consolidating all of the calls to close the db and cleanup the dir here.
```suggestion
	t.Cleanup(func() {
		require.NoError(t, pb.Close())
		require.NoError(t, os.RemoveAll(dbDir))
	}
	return pb, dbDir
```